### PR TITLE
Allow other database than mysql

### PR DESCRIPTION
**Pull request by flobz**
This allow to specify a full springDatasourceUrl instead of create url from springDatasourceHost and springDatasourceDb.

### CODE REVIEW

**Comment by flobz** [comment]
I updated chart version, workflow should pass.

File: charts/hawkbit/templates/deployment.yaml [hunk 1]
```diff
@@ -37,7 +37,7 @@ spec:
             - name: SPRING_PROFILES_ACTIVE
               value: "{{ .Values.spring.profiles }}"
             - name: "SPRING_DATASOURCE_URL"
-              value: "jdbc:mysql://{{ if .Values.mysql.enabled }}{{ .Release.Name }}-mysql{{ else }}{{ .Values.env.springDatasourceHost }}{{ end }}:3306/{{ .Values.env.springDatasourceDb }}"
+              value: "{{ if .Values.env.springDatasourceUrl }}{{ .Values.env.springDatasourceUrl }}{{ else }}jdbc:mysql://{{ if .Values.mysql.enabled }}{{ .Release.Name }}-mysql{{ else }}{{ .Values.env.springDatasourceHost }}{{ end }}:3306/{{ .Values.env.springDatasourceDb }} {{end}}"
```
Comment (calohmn) [inline charts/hawkbit/templates/deployment.yaml:40]: An extra space character got added here in the last part of the string:
`[...].springDatasourceDb }} {{end}}"`.

In general, this long line with the 2 `if` conditions is quite hard to read.
How about making multiple lines out of it?
E.g.
```
              {{- if .Values.env.springDatasourceUrl }}
              value: "{{ .Values.env.springDatasourceUrl }}"
              {{- else }}
              value: "jdbc:mysql://{{ if .Values.mysql.enabled }}{{ .Release.Name }}-mysql{{ else }}{{ .Values.env.springDatasourceHost }}{{ end }}:3306/{{ .Values.env.springDatasourceDb }}"
              {{- end}}
```

File: charts/hawkbit/values.yaml [hunk 2]
```diff
@@ -83,6 +83,7 @@ fileStorage:
 env:
   springDatasourceHost: "hawkbit-mysql"
   springDatasourceDb: "hawkbit"
+  springDatasourceUrl: ""
```
Comment (calohmn) [inline charts/hawkbit/values.yaml:86]: I think a comment would be helpful here, pointing out that if `springDatasourceUrl` is set, it takes precedence over the URL created from default mysql DB host (if `mysql.enabled`) or `springDatasourceHost` and `springDatasourceDb`.

File: charts/hawkbit/templates/deployment.yaml (hunk 1 above)
Comment (flobz) [reply charts/hawkbit/templates/deployment.yaml]: done

**Comment by calohmn** [comment]
@strailov Any comments on this?

**Comment by strailov** [comment]
Nope, looks good to me.

File: charts/hawkbit/templates/deployment.yaml [hunk 3]
```diff
@@ -37,7 +37,11 @@ spec:
             - name: SPRING_PROFILES_ACTIVE
               value: "{{ .Values.spring.profiles }}"
             - name: "SPRING_DATASOURCE_URL"
+              {{- if .Values.env.springDatasourceUrl }}
+              value: "{{ .Values.env.springDatasourceUrl }}"
+              {{ else }}
```
Comment (avgustinmm) [inline charts/hawkbit/templates/deployment.yaml:42]:
```suggestion
              {{- else }}
```
if you have set env.springDatasourceUrl then in the result deployment file you will have an empty line after "value: ...". 
better to change it to "{{- else }}" in order to remove the excessive empty line

**Comment by flobz** [comment]
@avgustinmm I apply your suggestion, can we merge ?

**Comment by avgustinmm** [comment]
I'm not a committer in the project. 
I haven't checked the PR extensively - just commented about something I've seen on first look.

**Review by calohmn — APPROVED** [review]
LGTM, thanks for contributing.

**Pull request merged**